File: CS/CS/Models/ModelValidationData.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace CS.Models {
    public class ModelValidationData {
        [Display(Name = "Name:")]
        [Required(ErrorMessage = "Name is required")]
        [StringLength(50, ErrorMessage = "Must be under 50 characters")]
        public string Name { get; set; }

        [Display(Name = "Age:")]
        [Range(18, 100, ErrorMessage = "Must be between 18 and 100")]
        public int? Age { get; set; }

        [Display(Name = "Email:")]
        [Required(ErrorMessage = "Email is required")]
        [RegularExpression("\\w+([-+.']\\w+)*@\\w+([-.]\\w+)*\\.\\w+([-.]\\w+)*", ErrorMessage = "Email is invalid")]
        public string Email { get; set; }

        [Display(Name = "Arrival Date:")]
        [Required(ErrorMessage = "Arrival date is required")]
        public DateTime? ArrivalDate { get; set; }
    }
}

File: VB/CS/Models/ModelValidationData.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.Linq
Imports System.Web

Namespace CS.Models
    Public Class ModelValidationData
        <Display(Name := "Name:"), Required(ErrorMessage := "Name is required"), StringLength(50, ErrorMessage := "Must be under 50 characters")> _
        Public Property Name() As String

        <Display(Name := "Age:"), Range(18, 100, ErrorMessage := "Must be between 18 and 100")> _
        Public Property Age() As Integer?

        <Display(Name := "Email:"), Required(ErrorMessage := "Email is required"), RegularExpression("\w+([-+.']\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*", ErrorMessage := "Email is invalid")> _
        Public Property Email() As String

        <Display(Name := "Arrival Date:"), Required(ErrorMessage := "Arrival date is required")> _
        Public Property ArrivalDate() As Date?
    End Class
End Namespace